### DISCUSSION TITLE
Large Samples (Video) over DDS

### DISCUSSION THREAD

**Discussion by 1GiLu1** [OpenDDS/OpenDDS#4853] [Q&A] [answered]
Hello,

I am trying to send an uncompressed Video over openDDS. So i am sending large samples with around 700.000 Bytes per sample and do this around 40 times a second.  
My .idl looks like this:
```   
@topic 
struct C_Video_Frame {  
long id;//@key
long frameNumber; 
sequence <char, 700000> frameData;
};
```
My DataWriter of this Topic is on PC1, the Datareader on PC2. Both PCs are connected directly via an ethernet cable. I started with standard QoS settings and basic configuration.

When i use the tcp transport for communication everything works fine, i have no sample loss and i can show the video on my receiver.

But when is use the rtps transport (which i want to use for my project), the DataWriter seems to write the samples correctly as i can verify on wireshark. But on the DataReader i lose a lot of Frames (around every fifth Frame, sometimes some frames in a row), which causes the video to be "laggy" on my receiver.

[DataReaderOutput.txt](https://github.com/user-attachments/files/17851432/DataReaderOutput.txt)
I attached a log output with highest Debug Level of the Subscriber which shows, that the "Refragmentation" on receiver side might be a problem. 
For samples with sequence number 659 all fragments are received and i get the "on_data_available" callback. For number 657 and 658 it receives the first fragments but then there is no debug output for the last fragments.

I already tried the following things:

- Increasing rcv_buffer_size in .ini File

- Use History QoS (KEEP_ALL and KEEP_LAST with higher depth)

I have to use OpenDDS 3.19 right now, so unfortunately i am not using the newest version.

If anybody has experience with this or ideas for improvement or better configuration (e.g. QoS), i would be very thankful !

**Comment by jrw972** [comment, marked as answer]
If you are using the default qos, then your reader is not reliable so you might want to try a reliable reader.
Without reliability, a single lost fragment is going to cause the entire sample to be discarded.
You will need to tune the reliability parameters to match your expected frame rate.

Increasing rcv_buffer_size is a good start.
I would look at the operating system itself.
For example, in Linux, you can see if you are dropping packets because the OS buffers are too small so you may need to resize those as well.

**Comment by 1GiLu1** [comment]
Hey Justin,

Thanks for your answer !

The Problem was the small Linux Receive Buffers. The default value ist just 213KB.  
If anybody else running into this problem:
You can just increase it with the following command to 25MB:

```
$ sudo sysctl -w net.core.rmem_max=26214400
net.core.rmem_max = 26214400
$ sudo sysctl -w net.core.rmem_default=26214400
net.core.rmem_default = 26214400
```

Just tested it via localhost until now, but here it solves the problem. Zero Images lost, even with Best-Effort Reliability. 
Thanks a lot!
